modules/ocf_postgres/manifests/init.pp

```diff
@@ -0,0 +1,23 @@
+class ocf_postgres {
+  class { 'postgresql::server':
+    postgres_password => hiera('postgres::root'),
+    #                      type    db       usr srcaddr   auth
+    ipv4acls          => ['hostssl sameuser all 0.0.0.0/0 md5'],
+    ipv6acls          => ['hostssl sameuser all ::/0 md5'];
```

Comment:
Could we use something more secure than md5 here? I think specifying `scram-sha-256` would be better, since using `md5` means using either `md5` or `scram-sha-256` but we don't really want to use md5 anyway.

Also, hostssl is probably fine, but we need to add a cert to this machine using the LE stuff if that's the case, otherwise connections won't be trusted correctly.

The IP ranges here might be better off as restricted to the subnet of the OCF (and localhost?) since that gives an extra layer if protection against outside connections if our external firewall doesn't work (unless you're planning on making this available to outside the subnet?). There's a nice option for that called `samenet` apparently. Alsocording to the postgresql docs, `::0/0` is supposed to be the IPv6 catch-all, but I'm not sure that extra 0 matters or not)

Reply:
I don't think `scram-sha-256` works before Postgres 10, unfortunately, and we're on 9.6.

I've added the ssl cert stuff to puppet, but Postgres needs the key to be owned by `postgres` or `root`, where it is currently owned by `ocfletsencrypt`. I'll try to find a solution to this.

I was planning on leaving it open to the world as I think OCF members ought to be able to use our db services outside of the context of a webapp running on our infra, like I was using OCF mysql as a backend for storing analysis data from research.